### PR TITLE
Add missing contract unit tests

### PR DESCRIPTION
**Pull request by RafilxTenfen**
### Add unit tests

Add more tests for NFT bridge

### Description

- Remove unused parts of contracts 
- Add unit tests for missing coverages

### How to Test

- Run both `ganache-cli`
- Run the integration tests

#### Case 1

1. Go to bridge directory
```shell
$~ cd bridge
```

2. Run the first ganache-cli
```shell
$~ npm run ganache
```

3. Open another shell and run the ganache mirror
```shell
$~ npm run ganache-mirror
```

4. Deploy the contracts to both chains
```shell
$~ npm run deployIntegrationTest
```

5. Go to the federator directory
```shell
$~ cd ../federator
```

6. Run the Integration tests
```shell
$~ npm run integrationTest
```
__Expected Result__
- It should pass the test
image

#### Bridge Directory `cd bridge`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Contracts are compiling `npm run compile`

#### Federator Directory `cd federator`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Typescript is compiling `npm run build`

### CODE REVIEW

File: bridge/contracts/nftbridge/SideNFTToken.sol [hunk 1]
```diff
@@ -9,103 +9,32 @@ import "../lib/LibEIP712.sol";
...
-        } else {
-            address recoveredAddress = ecrecover(digest, v, r, s);
-            require(recoveredAddress != address(0), "Invalid signature");
-            require(recoveredAddress == owner, "Unauthorized");
-        }
-
-        approve(spender, tokenId);
-    }
-
+  using Address for address;
```
Comment (pmprete) [inline bridge/contracts/nftbridge/SideNFTToken.sol:12]: we can remove the Address library as we are not using any more, actually we could do the same with SafeMath

File: bridge/contracts/test/nftbridge/TestTokenCreator.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+
+contract TestTokenCreator {
```
Comment (marcciosilva) [inline bridge/contracts/test/nftbridge/TestTokenCreator.sol:4]: couldn't we just use the NFTERC721TestToken contract for this?

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 3]
```diff
@@ -121,6 +122,46 @@ contract("Bridge NFT", async function(accounts) {
       assert.equal(result, "v1");
     });
 
+    it("should call the token creator", async function () {
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:125]: I'd maybe rename to "should return expected creator when calling getTokenCreator"

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 4]
```diff
@@ -121,6 +122,46 @@ contract("Bridge NFT", async function(accounts) {
       assert.equal(result, "v1");
     });
 
+    it("should call the token creator", async function () {
+      const testTokenCreator = await TestTokenCreator.new({ from: anAccount });
+      const tokenCreator = await this.NFTBridge.getTokenCreator(testTokenCreator.address, defaultTokenId);
+      assert.equal(tokenCreator, anAccount);
+    });
+
+    it("should change the Allow Tokens", async function() {
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:131]: I'd maybe rename to "should change the allow tokens as expected when calling changeAllowTokens"

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 5]
```diff
@@ -121,6 +122,46 @@ contract("Bridge NFT", async function(accounts) {
...
+      const newAllowTokens = await AllowTokens.new();
+      const receipt = await this.NFTBridge.changeAllowTokens(newAllowTokens.address, { from: bridgeManager });
+      truffleAssert.eventEmitted(receipt, "AllowTokensChanged", (ev) => {
+        return (
+          ev._newAllowTokens == newAllowTokens.address
+        );
+      });
+    });
+
+    it("should set the NFT bridge as upgrade", async function() {
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:141]: I'd maybe rename to "should emit 'Upgrading' event when calling setUpgrading"

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 6]
```diff
@@ -121,6 +122,46 @@ contract("Bridge NFT", async function(accounts) {
...
+      const receipt = await this.NFTBridge.setUpgrading(true, { from: bridgeManager });
+      truffleAssert.eventEmitted(receipt, "Upgrading", (ev) => {
+        return (
+          ev._isUpgrading
+        );
+      });
+    });
+
+    it("should change the Side Token Factory", async function() {
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:150]: I'd maybe rename to "should emit 'SideTokenFactoryChanged' event when calling changeSideTokenFactory"

**Review by pmprete — APPROVED** [review]
Except my comment looks good

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 7]
```diff
@@ -121,6 +122,46 @@ contract("Bridge NFT", async function(accounts) {
...
+      const receipt = await this.NFTBridge.changeSideTokenFactory(newSideTokenFactory.address, { from: bridgeManager });
+      truffleAssert.eventEmitted(receipt, "SideTokenFactoryChanged", (ev) => {
+        return (
+          ev._newSideNFTTokenFactory == newSideTokenFactory.address
+        );
+      });
+    });
+
+    it("should get the received selector", async function() {
+      const receipt = await this.NFTBridge.onERC721Received(bridgeOwner, tokenOwner, defaultTokenId, 0);
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:161]: this tests something from `IERC721Receiver`, do we need it?
besides, we only check that the transaction was successful, nothing more

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 8]
```diff
@@ -316,6 +357,11 @@ contract("Bridge NFT", async function(accounts) {
           });
           utils.checkRcpt(receipt);
 
+          const setFixedFee = await this.NFTBridge.getFixedFee();
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:360]: maybe I'd rename to `currentFixedFee` to not confuse this with a setter (somehow)

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 9]
```diff
@@ -767,6 +813,52 @@ contract("Bridge NFT", async function(accounts) {
...
+            event._sender === anAccount &&
+            event._tokenId.eq(new BN(tokenId)) &&
+            event._blockHash === blockHash &&
+            event._logIndex.eq(new BN(logIndex)) &&
+            event._receiver === anAccount
+          );
+        });
+      });
+
+      it("should check if it was claimed without claiming", async function () {
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:840]: I'd rename to ""should check if it was claimed without claiming when calling hasBeenClaimed"

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 10]
```diff
@@ -767,6 +813,52 @@ contract("Bridge NFT", async function(accounts) {
...
+        const transactionDataHash = await this.NFTBridge.getTransactionDataHash(anotherAccount, anAccount, tokenId, tokenAddress, blockHash, transactionHash, logIndex);
+        const result = await this.NFTBridge.hasBeenClaimed(transactionDataHash);
+        assert(!result, "The token should not been claimed yet");
+      });
+
+      it("should check if it was claimed successfully", async function () {
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:846]: this test is the opposite of the previous one (it tests that the negation of the assertion of the previous test is falsy), I think we should remove one of them)

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 11]
```diff
@@ -767,6 +813,52 @@ contract("Bridge NFT", async function(accounts) {
...
+            event._logIndex.eq(new BN(logIndex)) &&
+            event._receiver === anAccount
+          );
+        });
+      });
+
+      it("should check if it was claimed without claiming", async function () {
+        const transactionDataHash = await this.NFTBridge.getTransactionDataHash(anotherAccount, anAccount, tokenId, tokenAddress, blockHash, transactionHash, logIndex);
+        const result = await this.NFTBridge.hasBeenClaimed(transactionDataHash);
+        assert(!result, "The token should not been claimed yet");
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:843]: why do we use `!result`? this will evaluate the result string and become `false` if the string is empty for example (but maybe I'm wrong)
maybe what you meant was `assert.notEqual(result, "The token should not been claimed yet");`?

File: bridge/test/nftbridge/NFTBridge_test.js [hunk 12]
```diff
@@ -767,6 +813,52 @@ contract("Bridge NFT", async function(accounts) {
...
+
+      it("should check if it was claimed successfully", async function () {
+        await assertTokenIsLockedInBridge(this.NFTBridge.address, anotherAccount, this.token);
+        await claimTokenFromBridgeEnsuringEventEmission(this.sideNFTBridge, anotherAccount, anAccount, tokenId,
+            tokenAddress, blockHash, transactionHash, logIndex);
+        const result = await this.sideNFTBridge.hasBeenClaimed(transactionHash);
+        assert(result, "The token should have been claimed");
+      });
+
+      it("should check if it has crossed", async function () {
```
Comment (marcciosilva) [inline bridge/test/nftbridge/NFTBridge_test.js:854]: I'd add "... when calling hasCrossed"

File: bridge/test/nftbridge/SideNFTToken_test.js [hunk 13]
```diff
@@ -1,6 +1,8 @@
 const SideToken = artifacts.require("./SideNFTToken");
 const utils = require("../utils");
 const truffleAssert = require("truffle-assertions");
+const keccak256 = web3.utils.keccak256;
+const ethUtil = require('ethereumjs-util');
```
Comment (marcciosilva) [inline bridge/test/nftbridge/SideNFTToken_test.js:5]: we're not using these right? I'd maybe remove them

File: bridge/contracts/test/nftbridge/TestTokenCreator.sol (hunk 2 above)
Comment (RafilxTenfen) [reply bridge/contracts/test/nftbridge/TestTokenCreator.sol:4]: We could, but which address should be returned? The creator() function doesn't have tokenId as a parameter 
Do you think we could use the `_msgSender` as the returned value?

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 7 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js:161]: Well if we want to implement the interface `IERC721Receiver` we need to implement the method `onERC721Received`, and the test just check if it's not broken :/

File: bridge/contracts/test/nftbridge/TestTokenCreator.sol (hunk 2 above)
Comment (RafilxTenfen) [reply bridge/contracts/test/nftbridge/TestTokenCreator.sol:4]: Actually, we need another contract because we want to test the other edge from the `getTokenCreator` function

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 11 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js:843]: test removed, as the previous comment

File: bridge/contracts/nftbridge/SideNFTToken.sol (hunk 1 above)
Comment (RafilxTenfen) [reply bridge/contracts/nftbridge/SideNFTToken.sol]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 3 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 4 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 5 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 6 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 8 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 9 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 10 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: removed the first one at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/NFTBridge_test.js (hunk 12 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/NFTBridge_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

File: bridge/test/nftbridge/SideNFTToken_test.js (hunk 13 above)
Comment (RafilxTenfen) [reply bridge/test/nftbridge/SideNFTToken_test.js]: done at https://github.com/rsksmart/tokenbridge/pull/268/commits/9c4e1c7b64b7379ff804bb3f768eed9c96a7461b

**Review by marcciosilva — APPROVED** [review]
